source/time_stepping/function.cc

```diff
@@ -30,7 +30,8 @@ namespace aspect
     double
     Function<dim>::execute()
     {
-      double new_time_step = function.value(Point<1>(this->get_time()));
+      const double time_scale = this->convert_output_to_years() ? constants::year_in_seconds : 1.0;
+      double new_time_step    = function.value(Point<1>(this->get_time()) / time_scale) * time_scale;
```

Comment:
The new time step can be marked const as well:

```suggestion
      const double time_scale    = this->convert_output_to_years() ? constants::year_in_seconds : 1.0;
      const double new_time_step = function.value(Point<1>(this->get_time()) / time_scale) * time_scale;
```